src/components/TimeControl/util.ts

```diff
@@ -408,27 +406,42 @@ export function shortShortTimeControl(time_control) { /* {{{ */
     }
 }  /* }}} */
 
+const QUESTIONABLE_SECONDS_PER_MOVE = 4;  // less than this gets flagged as may be cheaty.
+const QUESTIONABLE_ABSOLUTE_TIME = 900; // Arguably absolute time cheaters don't use > 10 min.  I've seen reports complaining about abuse at 10min though, so set this a bit higher.
+
 export function usedForCheating(time_control) {
     if (typeof(time_control) !== "object" || time_control === null) {
         return false;
     }
 
+    // either there has to be enough time for the whole game or
+    // a sensible ongoing per-move allocation
     switch (time_control.system || time_control.time_control) {
 
         case "simple":
             return time_control.per_move < QUESTIONABLE_SECONDS_PER_MOVE;
 
         case "absolute":
-            return time_control.total_time <= 900; // Arguably absolute time cheaters don't use > 10 min.  I've seen reports complaining about abuse at 10min though, so set this a bit higher.
+            return time_control.total_time <= QUESTIONABLE_ABSOLUTE_TIME;
 
         case "canadian":
-            return time_control.main_time < QUESTIONABLE_SECONDS_PER_MOVE && time_control.period_time / time_control.stones_per_period < QUESTIONABLE_SECONDS_PER_MOVE;
+            return !(
+                time_control.main_time > QUESTIONABLE_ABSOLUTE_TIME ||
+                time_control.period_time / time_control.stones_per_period > QUESTIONABLE_SECONDS_PER_MOVE
+            );
 
         case "byoyomi":
-            return time_control.main_time < QUESTIONABLE_SECONDS_PER_MOVE && time_control.period_time < QUESTIONABLE_SECONDS_PER_MOVE;
+            return !(
+                time_control.main_time > QUESTIONABLE_ABSOLUTE_TIME ||
+                time_control.period_time > QUESTIONABLE_SECONDS_PER_MOVE
+            );
 
-        // haven't seen silliness, not sure about sensible criteria
         case "fischer":
+            return !(
+                time_control.initial_time > QUESTIONABLE_ABSOLUTE_TIME ||
```

Comment:
I've seen a lot of good fisher games that have less than 15 minutes initial time, it's less important so long as the time increment is reasonable - on the flip side being able to build up to 15 minutes and and then wait will definitely not sit well with some people. Maybe something more like 30 seconds as a minimum?